app/jobs/ingest_archival_media_bag_job.rb

```diff
@@ -66,22 +66,41 @@ def ingest
         component_groups.each do |cid, sides|
           media_resource = find_or_create_media_resource(cid)
           media_resource_change_set = MediaResourceChangeSet.new(media_resource, source_metadata_identifier: media_resource.source_metadata_identifier.first)
+          add_av(media_resource_change_set, sides)
+          add_pbcore(media_resource_change_set, sides)
+          media_resource_change_set.member_of_collection_ids += Array.wrap(collection.id)
+          changeset_persister.save(change_set: media_resource_change_set)
+        end
+      end
+
+      private
+
+        def add_av(media_resource_change_set, sides)
           sides.each do |side|
             file_set = create_av_file_set(side)
             media_resource_change_set.member_ids += Array.wrap(file_set.id)
             media_resource_change_set.sync
           end
-          media_resource_change_set.member_of_collection_ids += Array.wrap(collection.id)
-          changeset_persister.save(change_set: media_resource_change_set)
         end
-      end
 
-      private
+        def add_pbcore(media_resource_change_set, sides)
+          sides.map { |side| side.split("_").first }.uniq.each do |barcode|
+            file_set = create_pbcore_file_set(barcode)
+            media_resource_change_set.member_ids += Array.wrap(file_set.id)
+            media_resource_change_set.sync
+          end
+        end
 
-        # Retrieve a Hash of EAD Component IDs/Barcodes for file barcodes specified in a given Bag
-        # @return [Hash] map of EAD component IDs to file barcodes
-        def component_groups
-          @component_groups ||= bag.component_groups
+        # Creates and persists a FileSet for a pbcore xml file
+        # @param barcode [String] barcode_side for a given media object
+        # @return [FileSet] the persisted FileSet containing the binary and file metadata
+        def create_pbcore_file_set(barcode)
+          file_set = FileSet.new(title: barcode)
+          pbcore = bag.pbcore_parsers.find { |pbcore_parser| pbcore_parser.barcode == barcode }
+          file = IngestableFile.new(file_path: pbcore.path, mime_type: "application/xml; schema=pbcore", original_filename: pbcore.path.basename)
```

Comment:
To avoid demeter violations can we have `pbcore.original_filename` or something similar?